From 4a82094c9822a989722acf30fff9679a9ad1a311 Mon Sep 17 00:00:00 2001
From: Kostadin Shishmanov <kocelfc@tutanota.com>
Date: Sat, 30 Dec 2023 22:47:38 +0200
Subject: [PATCH] Fix build with musl

Fixes the following build failure:
In file included from ../common/common-cpu.c:32:
../common/common-cpu.h:44:9: error: unknown type name 'cpu_set_t'
   44 |         cpu_set_t *online;
      |         ^~~~~~~~~
../common/common-cpu.h:45:9: error: unknown type name 'cpu_set_t'
   45 |         cpu_set_t *to_keep;
      |         ^~~~~~~~~

Signed-off-by: Kostadin Shishmanov <kocelfc@tutanota.com>
---
 common/common-cpu.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/common/common-cpu.h b/common/common-cpu.h
index 3c8df27c..ccc8ed44 100644
--- a/common/common-cpu.h
+++ b/common/common-cpu.h
@@ -31,6 +31,8 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #pragma once
 
+#define _GNU_SOURCE
+
 #include <sched.h>
 #include <stdlib.h>
 
